nova_vm/src/ecmascript/builtins/control_abstraction_objects/promise_objects/promise_constructor.rs

```diff
@@ -102,20 +174,34 @@ impl PromiseConstructor {
     fn race(_agent: &mut Agent, _this_value: Value, _arguments: ArgumentsList) -> JsResult<Value> {
         todo!()
     }
-    fn reject(
-        _agent: &mut Agent,
-        _this_value: Value,
-        _arguments: ArgumentsList,
-    ) -> JsResult<Value> {
-        todo!()
+
+    fn reject(agent: &mut Agent, _this_value: Value, arguments: ArgumentsList) -> JsResult<Value> {
+        // 1. Let C be the this value.
+        // 2. Let promiseCapability be ? NewPromiseCapability(C).
+        let promise_capability = PromiseCapability::new(agent);
+        // 3. Perform ? Call(promiseCapability.[[Reject]], undefined, « r »).
+        promise_capability.reject(agent, arguments.get(0));
+        // 4. Return promiseCapability.[[Promise]].
```

Comment:
**thought:** The spec does a lot of unnecessary things here in the usual case.

I presume in the case of actual `Promise.reject(new Error())` the only thing you _really_ need to do is create a new Promise with its internal state as rejected, right?

Reply:
Now that you mention it, yeah, this could be simplified. In fact, *in every case* the only thing you'd have to do is create an already rejected promise. This will break down with promise subclasses, of course, but for now we're not caring at all about them.